cluster-autoscaler/cluster_autoscaler.go

```diff
@@ -410,7 +380,7 @@ func main() {
 		kube_leaderelection.RunOrDie(kube_leaderelection.LeaderElectionConfig{
 			Lock: &resourcelock.EndpointsLock{
 				EndpointsMeta: apiv1.ObjectMeta{
-					Namespace: "kube-system",
```

Comment:
Yeah, lets fix it. Provide a flag with namespace and populate it via downward api. The flag is useful in this PR anyway.